modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/DeliverableAndesMetadata.java

```diff
@@ -474,17 +474,26 @@ private boolean isMarkAsAcked() {
         for (Map.Entry<UUID, ChannelInformation> channelInfoEntry : channelDeliveryInfo.entrySet()) {
             ChannelMessageStatus messageStatus = channelInfoEntry.getValue().getLatestMessageStatus();
 
-            //if channel is closed ignore it from considering
-            if (null != messageStatus && messageStatus.equals(ChannelMessageStatus.CLOSED)) {
-                continue;
-            }
-            //if message is rejected by client repeatedly ignore it from considering
-            if (null != messageStatus && messageStatus.equals(ChannelMessageStatus.CLIENT_REJECTED)) {
-                continue;
-            }
-            if (null == messageStatus || !messageStatus.equals(ChannelMessageStatus.ACKED)) {
+            if (null == messageStatus) {
```

Comment:
There is. 

ChannelInformation object has the list of channel message status. ChannelInformation is created when a message is "scheduled" for a subscriber. First status added is "DISPATCHED" when disruptor write message to wire. Thus when a message is in the disruptor channel message status list is empty. Thus null check is needed.